interpreter/CMakeLists.txt

```diff
@@ -134,24 +160,243 @@ set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES "")
 set(LLVM_APPEND_VC_REV OFF CACHE BOOL "Disabled LLVM revision dependency" FORCE)
 
 #---Add the sub-directory excluding all the targets from all-----------------------------------------
-if(CMAKE_GENERATOR MATCHES "Xcode")
-  add_subdirectory(llvm/src)
+if(builtin_llvm)
+  if(CMAKE_GENERATOR MATCHES "Xcode")
+    add_subdirectory(llvm/src)
+  else()
+    add_subdirectory(llvm/src EXCLUDE_FROM_ALL)
+  endif()
+
+  #---Make LLVM compilation flags public---------------------------------------------------------------
+  # Extract the compilation flags from LLVM and make them public to the
+  # rest of ROOT so that we can compile against LLVM with matching flags.
+
+  # LLVM doesn't really give us a API to get this with an in-source build
+  # so we just use the normal way of doing this and read the llvm directory
+  # compilation properties.
```

Comment:
This is strange, the flags should come with the targets already, no? If you use `target_link_libraries()`, any flags needed will be automatically appended (i.e. include directories and compile flags). This seems a bit hacky.

Reply:
That's legacy code, I moved it and indented it, that's why it shows up here. I do not know how to solve this at the moment.

Reply:
Ok, let's leave it for later than.